src/App.tsx

```diff
@@ -15,48 +17,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+function changeSorting(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const newGoods = [...goods];
+
+  if (sortType !== SortType.NONE) {
+    newGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    newGoods.reverse();
+  }
+
+  return newGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversity] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const visibleGoods = changeSorting(goodsFromServer, sortType, isReversed);
+
+  const reset = () => {
+    setSortType(SortType.NONE);
+    setReversity(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button',
+            'is-info',
+            {
+              'is-light': sortType !== SortType.ALPHABET,
+            },
+          )}
+          onClick={() => setSortType(SortType.ALPHABET)}
```

Comment:
Create separate methods instead of creating inline event handlers.
<img width="1058" alt="Screenshot 2022-09-14 at 12 19 46" src="https://user-images.githubusercontent.com/17098736/190176837-ad56985e-a3dd-4046-bc3f-372f7bade9a5.png">